include/reshade_api_display.hpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2024 Patrick Mours
+ * SPDX-License-Identifier: BSD-3-Clause OR MIT
+ */
+
+#pragma once
+
+#include "reshade_api_device.hpp"
+#include <unordered_map>
+#include <memory>
+
+namespace reshade { namespace api
+{
+	/// <summary>
+	/// Describes quantities defined precisely as the ratio of two integers, such as refresh rates.
+	/// </summary>
+	struct rational
```

Comment:
For consistency with other parts of the API, I'd forego this type. `swapchain_desc::fullscreen_refresh_rate` e.g. is a floating point value describing the refresh rate, not a rational type (and there is some logic at https://github.com/crosire/reshade/blob/main/source/dxgi/dxgi.cpp#L21 to convert between the two), so it would be strange if the display type suddenly expressed the refresh rate in a different manner.